swarm/storage/mru/resource.go

```diff
@@ -800,35 +808,42 @@ func (h *Handler) BlockToPeriod(name string, blocknumber uint64) (uint32, error)
 func (h *Handler) PeriodToBlock(name string, period uint32) uint64 {
 	return h.resources[name].startBlock + (uint64(period) * h.resources[name].frequency)
 }
+**/
 
 // Retrieves the resource index value for the given nameHash
-func (h *Handler) get(nameHash string) *resource {
+func (h *Handler) get(rootAddr storage.Address) *resource {
+	if rootAddr == nil {
+		log.Warn("Handler.get with nil rootAddr")
+		return nil
+	}
+	hashKey := *(*uint64)(unsafe.Pointer(&rootAddr[0]))
 	h.resourceLock.RLock()
 	defer h.resourceLock.RUnlock()
-	rsrc := h.resources[nameHash]
+	rsrc := h.resources[hashKey]
 	return rsrc
 }
 
 // Sets the resource index value for the given nameHash
-func (h *Handler) set(nameHash string, rsrc *resource) {
+func (h *Handler) set(rootAddr storage.Address, rsrc *resource) {
+	hashKey := *(*uint64)(unsafe.Pointer(&rootAddr[0]))
 	h.resourceLock.Lock()
 	defer h.resourceLock.Unlock()
-	h.resources[nameHash] = rsrc
+	h.resources[hashKey] = rsrc
 }
 
 // used for chunk keys
-//func (h *Handler) resourceHash(period uint32, version uint32, nameHash common.Hash, publicKeyBytes [publicKeyLength]byte) storage.Address {
-func (h *Handler) resourceHash(period uint32, version uint32, nameHash common.Hash, ownerAddr common.Address) storage.Address {
-	hasher := h.hashPool.Get().(storage.SwarmHash)
-	defer h.hashPool.Put(hasher)
+func resourceHash(period uint32, version uint32, rootAddr storage.Address) storage.Address {
+	hasher := hashPool.Get().(storage.SwarmHash)
+	defer hashPool.Put(hasher)
 	hasher.Reset()
-	hasher.Write(NewResourceHash(period, version, nameHash, ownerAddr))
+	hasher.Write(NewResourceHash(period, version, rootAddr))
```

Comment:
hash a "new resourcehash" to get a "resourcehash" suggests slightly misleading namings

Reply:
This is refactored in #704 as follows:
* New structure to hold the lookup components, `UpdateLookup` which contains `version`, `period`, `rootAddr`.
* `resourcehash` -> `UpdateLookup.GetUpdateAddr()`
* `NewResourceHash` is gone since the structure serializer is reused.